### ISSUE TITLE
Improve workloads experience when customers are changing around feeds

### ISSUE THREAD

**Issue by marcpopMSFT** [dotnet/sdk#38784]
### Is your feature request related to a problem? Please describe.
Customers (especially 1P) often install workloads off of specific feeds. We've gotten feedback that they'll install with a feed and then later decided to add an additional workload or want to revert to a prior version.

Reverting to a prior version will be handled by #36809. However, we can improve the experience when working with multiple feeds.

### Describe the solution you'd like
When we install a manifest, we should record somewhere which feed was used to install that manifest.  The assumption here is that any feed that has a manifest also has the matching packs you'll need (need to confirm with workloads partners). This is not true for runtime workloads since the 8.0 branch contains manifests that point to specific 6.0 and 7.0 builds so something we'd need to consider.

When you try to install a workload and we fail to find the pack to install, that should cause us to wonder how we got in a state with a manifest that asks for a pack we don't have. If we have that feed stored, we should be able to suggest an option for the customer to use the feeds we saved off.  Think "dotnet workload install maui --use-cached-feeds" or something like that.

**One consideration**: Is it possible to use package source mapping for this? Workloads are all downloaded using nupkgs so it wasn't clear if we could save off a private package source mapping that we don't use except under certain flags. Noting that we would have the mapping for the manifest but not the packs in this situation so we'd really just want to pull the feeds out of the source mapping. @jeffkl for his thoughts.

**Comment by jeffkl** [comment]
I don't think this is how package source mapping works, the repo's NuGet.config specifies the mappings.  I'll include @nkolev92 for his input.

**Comment by nkolev92** [comment]
Yup, @jeffkl is spot on here. 
Source mapping decides where you download a package from. 
Basically, it only plays when choosing where to download a package from.
